### ISSUE TITLE
Crashing with error:  Error receiving broadcast Intent com.google.android.gms.auth.api.phone.SMS_RETRIEVED

### ISSUE THREAD

**Issue by jimmyff** [Tkko/Flutter_Pinput#105]
Hi, 
I'm getting multiple reports of my app crashing at the time they arrive on a screen that uses pinput. In crashalytics I see the following:
```
Fatal Exception: java.lang.RuntimeException: Error receiving broadcast Intent { act=com.google.android.gms.auth.api.phone.SMS_RETRIEVED flg=0x200010 pkg=com.onescene.app (has extras) } in e.f.a.c.e.h.mm@aa016e
       at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1816)
       at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(:2)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8855)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)


Caused by java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
       at java.util.regex.Matcher.reset(Matcher.java:256)
       at java.util.regex.Matcher.<init>(Matcher.java:167)
       at java.util.regex.Pattern.matcher(Pattern.java:1040)
       at e.f.a.c.e.h.om.b(:6)
       at e.f.a.c.e.h.mm.onReceive(:68)
       at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1806)
       at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(:2)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8855)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)

```
I'm using `androidSmsAutofillMethod: AndroidSmsAutofillMethod.smsUserConsentApi`, 

Is this pinput autofill causing the issue? I'm going to turn of autofill for now.

Thanks

**Comment by Tkko** [comment]
Hey @jimmyff, Could you tell me on which devices does crash happen? And are those crashes real ANRs (AKA Fatal crashes in Crachlytics) or just background error messages?

**Comment by jimmyff** [comment]
Hey thanks for the response. I had reports from OnePlus 7T and Galaxy Flip 4. Yes the whole app crashes/closes.

Thanks

**Comment by Tkko** [comment]
Omg that sucks, I'll dive into the issue ASAP

**Comment by Tkko** [comment]
Resolved in version [2.2.20](https://pub.dev/packages/pinput/changelog)

**Comment by guvanch07** [comment]
@Tkko , Hi
I got this error in version 3.0.1
but i solved it by setting ``` timeout: const Duration(seconds: 0)``` of FirebaseAuth.verifyPhoneNumber
